### PR TITLE
Added documentation warnings about empty clipped reads from readClipper where they can occur

### PR DESCRIPTION
**Pull request by jamesemery**
* Fixed an issue where OverhangFixingManger did not handle empty reads correctly resulting in an NPE
* Checked every codepath that could result in an empty read in readClipper and ensured that it handles empty reads.

Resolves: https://github.com/broadinstitute/gatk/issues/4204

### CODE REVIEW

**Comment by droazen** [comment]
@jamesemery This is failing with a compiler error:

```
error: package org.testng.annotations does not exist
import org.testng.annotations.Test
```

**Comment by jamesemery** [comment]
@droazen Fixed the error

File: src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSpark.java [hunk 1]
```diff
@@ -106,6 +107,8 @@
 
         // Here we combine the original bam with the repartitioned unmarked readnames to produce our marked reads
         return sortedReadsForMarking.zipPartitions(repartitionedReadNames, (readsIter, readNamesIter)  -> {
+            System.out.println("system property says: " + System.getProperty("samjdk.compression_level"));
+            System.out.println("htsjdk says: " + Defaults.COMPRESSION_LEVEL);
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSpark.java]: @jamesemery These debug print statements seem to have snuck into your latest commit -- can you remove them?

Comment (jamesemery) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSpark.java]: whoops, yeah they are gone

**Comment by jamesemery** [comment]
@droazen back to you

**Pull request merged**